exchanges/poloniex/poloniex.go

```diff
@@ -16,915 +17,1002 @@ import (
+		return nil, kline.ErrUnsupportedInterval
 	}
-
-	var resp []ChartData
-	err = json.Unmarshal(temp, &resp)
-	if err != nil {
-		var errResp struct {
-			Error string `json:"error"`
-		}
-		if errRet := json.Unmarshal(temp, &errResp); errRet != nil {
-			return nil, errRet
-		}
-		if errResp.Error != "" {
-			return nil, errors.New(errResp.Error)
+	params := url.Values{}
+	params.Set("interval", intervalString)
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	if !startTime.IsZero() && !endTime.IsZero() {
+		if err := common.StartEndTimeCheck(startTime, endTime); err != nil {
+			return nil, err
 		}
+		params.Set("startTime", strconv.FormatInt(startTime.UnixMilli(), 10))
+		params.Set("endTime", strconv.FormatInt(endTime.UnixMilli(), 10))
 	}
+	var resp []*CandlestickData
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, unauthEPL, common.EncodeURLValues(marketsPath+symbol.String()+"/candles", params), &resp)
+}
 
-	return resp, err
+// GetTrades returns a list of recent trades, request param limit is optional, its default value is 500, and max value is 1000.
+func (e *Exchange) GetTrades(ctx context.Context, symbol currency.Pair, limit int64) ([]*Trade, error) {
+	if symbol.IsEmpty() {
+		return nil, currency.ErrCurrencyPairEmpty
+	}
+	params := url.Values{}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	var resp []*Trade
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, referenceDataEPL, common.EncodeURLValues(marketsPath+symbol.String()+"/trades", params), &resp)
 }
 
-// GetCurrencies returns information about currencies
-func (e *Exchange) GetCurrencies(ctx context.Context) (map[string]*Currencies, error) {
-	type Response struct {
-		Data map[string]*Currencies
+// GetTickers retrieve ticker in last 24 hours for all symbols.
+func (e *Exchange) GetTickers(ctx context.Context) ([]*TickerData, error) {
+	var resp []*TickerData
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, referenceDataEPL, "/markets/ticker24h", &resp)
+}
+
+// GetTicker retrieve ticker in last 24 hours for provided symbols.
+func (e *Exchange) GetTicker(ctx context.Context, symbol currency.Pair) (*TickerData, error) {
+	if symbol.IsEmpty() {
+		return nil, currency.ErrCurrencyPairEmpty
 	}
-	resp := Response{}
-	return resp.Data, e.SendHTTPRequest(ctx,
-		exchange.RestSpot,
-		"/public?command=returnCurrencies&includeMultiChainCurrencies=true",
-		&resp.Data,
-	)
+	var resp *TickerData
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, referenceDataEPL, marketsPath+symbol.String()+"/ticker24h", &resp)
 }
 
-// GetTimestamp returns server time
-func (e *Exchange) GetTimestamp(ctx context.Context) (time.Time, error) {
-	var resp TimeStampResponse
-	err := e.SendHTTPRequest(ctx,
-		exchange.RestSpotSupplementary,
-		poloniexTimestamp,
-		&resp,
-	)
-	if err != nil {
-		return time.Time{}, err
+// GetCollateralInfo retrieves collateral information for currency
```

Comment:
🚧 Change request:
The change to this comment went from something kinda okay, to something worse, and I'm not sure why. `for all currencies` became `for currency`.
`for currencies` would have been better.
But my suggestion stands, better to just remove it, since it's implicitly currencies.
```suggestion
// GetCollateralInfo retrieves collateral information
```